src/tsickle_compiler_host.ts

```diff
@@ -155,71 +157,70 @@ export class TsickleCompilerHost implements ts.CompilerHost {
     this.delegate.writeFile(fileName, content, writeByteOrderMark, onError, sourceFiles);
   }
 
-  sourceMapConsumerToGenerator(sourceMapConsumer: SourceMapConsumer): SourceMapGenerator {
-    return SourceMapGenerator.fromSourceMap(sourceMapConsumer);
-  }
+  getSourceMapKeyForPathAndName(outputFilePath: string, sourceFileName: string): string {
+    const fileDir = path.dirname(outputFilePath);
 
-  /**
-   * Tsc identifies source files by their relative path to the output file.  Since
-   * there's no easy way to identify these relative paths when tsickle generates its
-   * own source maps, we patch them with the file name from the tsc source maps
-   * before composing them.
-   */
-  sourceMapGeneratorToConsumerWithFileName(
-      sourceMapGenerator: SourceMapGenerator, fileName: string): SourceMapConsumer {
-    const rawSourceMap = sourceMapGenerator.toJSON();
-    rawSourceMap.sources = [fileName];
-    rawSourceMap.file = fileName;
-    return new SourceMapConsumer(rawSourceMap);
+    return this.getCanonicalFileName(path.resolve(fileDir, sourceFileName));
   }
 
-  sourceMapTextToConsumer(sourceMapText: string): SourceMapConsumer {
-    const sourceMapJson: any = sourceMapText;
-    return new SourceMapConsumer(sourceMapJson);
+  getSourceMapKeyForSourceFile(sourceFile: ts.SourceFile): string {
+    return this.getCanonicalFileName(sourceFile.path);
   }
 
-  getSourceMapKey(outputFilePath: string, sourceFileName: string): string {
-    const fileDir = path.dirname(outputFilePath);
+  stripAndStoreExistingSourceMap(sourceFile: ts.SourceFile): void {
+    if (sm.containsInlineSourceMap(sourceFile.text)) {
+      const sourceMapJson = sm.extractInlineSourceMap(sourceFile.text);
+      const sourceMap = sm.sourceMapTextToGenerator(sourceMapJson);
+      this.preexistingSourceMaps.set(this.getSourceMapKeyForSourceFile(sourceFile), sourceMap);
 
-    return this.getCanonicalFileName(path.resolve(fileDir, sourceFileName));
+      sourceFile.text = sm.removeInlineSourceMap(sourceFile.text);
```

Comment:
It's not generally safe to mutate the TypeScript AST. We've run into very hard to track bugs because of it before. Can you extract the source map on load?